src/charm.py

```diff
@@ -415,7 +404,7 @@ def configure_bgp(self):
 
     def set_active_status(self):
         """Set active if cni is configured."""
-        if self.stored.tigera_cni_configured:
+        if cast(bool, self.stored.tigera_cni_configured):
```

Comment:
it feels like something meaningful is happening here.  What's the purpose of the `cast`